aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -391,6 +396,74 @@ impl AptosVM {
         }
         Ok(())
     }
+    fn verify_module_init_function(
```

Comment:
half related to this, we should think about how to enforce this type of checks on compile time, not runtime. we have special rules on script functions as well

Reply:
Asked in the move community meeting.  The feedbacks is `different chain may have different signature and is adapter specific`. seems the check should not be done at compiling time.

Reply:
we can have our wrapper around the move compiler to report errors

Reply:
make sense. I think users compile the move modules using Aptos Cli. If I add Aptos customized verification in our Cli, it may lead to inconsistent compiling results between Aptos Cli and Move Cli. Do we want a Aptos Move language?

Reply:
We already have that due to native functions.